### ISSUE TITLE
Auth.AwsLogin usage documentation incorrect

### ISSUE THREAD

**Issue by mschuchard** [hashicorp/vault-client-go#159]
# Current Behavior

Similar to #158, the documented usage of `Auth.AwsLogin` is incorrect:

```go
request := schema.NewAwsLoginRequestWithDefaults()
resp, err := client.Auth.AwsLogin(
	context.Background(),
	awsMountPath,
	request,
	vault.WithToken("my-token"),
)
```

# Failure Information

Type mismatches on the arguments as one would expect.
Version 0.3.0

# Additional Information

Corrected usage is:

```go
defaultRequest := schema.NewAwsLoginRequestWithDefaults()
response, err := client.Auth.AwsLogin(
  context.Background(),
  *defaultRequest,
  vault.WithMountPath(awsMountPath),
)
```

I have also seen #147 so I am aware I would need to generic `Write` or fallback to the `aws` subpackage of the `vault` module anyway.

**Comment by averche** [comment]
Thanks for creating this issue! I have addressed it partially in #161 with proper generation of mount-path related logic. I'm trying to use the `x-vault-unauthenticated` OpenAPI tag to not generate the authentication parts for the login requests but the template logic is  a bit tricky.

**Comment by maxb** [comment]
Once #223 is merged, the only remaining issue with this documentation will be that it is meaningless to suggest specifying a token, when making a login request.

I wonder if we might solve that by just removing any mention to creating a client, or setting a token, from the individual documentation for each method? After all, it is very repetitive.

Instead of starting off every per-method code snippet with
```
	client, err := vault.New(
		vault.WithAddress("http://127.0.0.1:8200"),
	)
	if err != nil {
		log.Fatal(err)
	}
```
we could just say
```
	client := /* a suitably configured *vault.Client - see top-level README.md */
```

and omit the

```
		vault.WithToken("my-token"),
```
line.

Any further elaboration is better written once, rather than being rendered 900+ times for every individual method.
